packages/yew/src/platform/pinned/mpsc.rs

```diff
@@ -0,0 +1,395 @@
+    type Item = T;
+
+    fn poll_next(
+        self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions and hence uniquely owns the
+        // mutable reference.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &mut *self.inner.get() };
+
+        match (inner.items.pop_front(), inner.closed) {
+            (Some(m), _) => Poll::Ready(Some(m)),
+            (None, false) => {
+                inner.rx_waker = Some(cx.waker().clone());
+                Poll::Pending
+            }
+            (None, true) => Poll::Ready(None),
+        }
+    }
+}
+
+impl<T> FusedStream for UnboundedReceiver<T> {
+    fn is_terminated(&self) -> bool {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions and hence uniquely owns the
+        // mutable reference.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &*self.inner.get() };
+        inner.items.is_empty() && inner.closed
+    }
+}
+
+impl<T> Drop for UnboundedReceiver<T> {
+    fn drop(&mut self) {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions and hence uniquely owns the
+        // mutable reference.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &mut *self.inner.get() };
+        inner.close();
+    }
+}
+
+/// The sender of an unbounded mpsc channel.
+#[derive(Debug)]
+pub struct UnboundedSender<T> {
+    inner: Rc<UnsafeCell<Inner<T>>>,
+}
+
+impl<T> UnboundedSender<T> {
+    /// Sends a value to the unbounded receiver.
+    pub fn send_now(&self, item: T) -> Result<(), SendError<T>> {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any function that have already acquired a mutable
+        // reference.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &mut *self.inner.get() };
+
+        if inner.closed {
+            return Err(SendError { inner: item });
+        }
+
+        inner.items.push_back(item);
+
+        if let Some(ref m) = inner.rx_waker {
+            m.wake_by_ref();
+        }
+
+        Ok(())
+    }
+
+    /// Closes the channel.
+    pub fn close_now(&self) {
```

Comment:
I don't see a reason for `send_now` and `close_now` to be `pub`, is there some call outside the code of this PR?

Reply:
Sink::close and Sink::send are async and can only be called from async fn. These 2 methods allow Senders to be used in non-async context.